### PR TITLE
Support android

### PR DESCRIPTION
**Pull request by andyLiuChina**
Compiling passed under the android platform, and write a demo named androidplay in the folder of examples, you can use the command "./androidplay xxx.264" to display.

### CODE REVIEW

File: examples/androidplayer.cpp [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+                break;
+
+            ANativeWindowBuffer* buf;
+            err = native_window_dequeue_buffer_and_wait(mNativeWindow.get(), &buf);
+            if (err != 0) {
+                fprintf(stderr, "dequeueBuffer failed: %s (%d)\n", strerror(-err), -err);
+                break;
+            }
+
+            std::map< ANativeWindowBuffer*, SharedPtr<VideoFrame> >::const_iterator it;
```
Comment (xuguangxin) [inline examples/androidplayer.cpp]: could we hid this in function convertToVideoFrame

File: examples/androidplayer.cpp [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+        VASurfaceAttrib attrib;
+        memset(&attrib, 0, sizeof(attrib));
+
+        VASurfaceAttribExternalBuffers external;
+        memset(&external, 0, sizeof(external));
+
+        external.pixel_format = VA_FOURCC_NV12;
+        external.width = w;
+        external.height = h;
+        external.pitches[0] = w; //?
```
Comment (xuguangxin) [inline examples/androidplayer.cpp]: could we query this from gralloc? instead use w.

File: examples/androidplayer.cpp [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+        err = native_window_set_buffer_count(mNativeWindow.get(), 5);
+        if (err != 0) {
+            ALOGE("native_window_set_buffer_count failed: %s (%d)", strerror(-err), -err);
+            return false;
+        }
+
+        return true;
+    }
+
+    SharedPtr<VideoFrame> createVaSurface(uint32_t w, uint32_t h, GraphicBuffer* buf)
```
Comment (xuguangxin) [inline examples/androidplayer.cpp]: could we use GraphicBuffer->getWidth() & GraphicBuffer->getWheight()

File: examples/androidplayer.cpp (hunk 2 above)
Comment (xuguangxin) [inline examples/androidplayer.cpp]: is it possible use GraphicBuffer->getPitch()?

**Comment by xuguangxin** [comment]
All commits pushed, except last two

**Comment by halleyzhao** [comment]
I want to give it a review, but the code base is old, and Guangxin's comments hasn't been followed yet.
I will review it after a update.

File: Android.mk [hunk 4]
```diff
@@ -8,4 +8,30 @@ include $(LIBYAMICODEC_PATH)/vaapi/Android.mk
 include $(LIBYAMICODEC_PATH)/decoder/Android.mk
 include $(LIBYAMICODEC_PATH)/encoder/Android.mk
 include $(LIBYAMICODEC_PATH)/vpp/Android.mk
+include $(LIBYAMICODEC_PATH)/v4l2/Android.mk
+
+include $(CLEAR_VARS)
+
+LOCAL_MULTILIB = 64
```
Comment (xuguangxin) [inline Android.mk]: why we need this?
It's mean we only build 64bit version?

File: examples/Android.mk [hunk 5]
```diff
@@ -23,14 +23,8 @@ LOCAL_SHARED_LIBRARIES := \
...
-        libstagefright_foundation
-
-LOCAL_WHOLE_STATIC_LIBRARIES := \
-        libyami_common \
-        libcodecparser \
-        libyami_vaapi \
-        libyami_decoder \
-        libyami_vpp
+        libyami \
+        libstagefright_foundation \
```
Comment (xuguangxin) [inline examples/Android.mk]: why we need this?

File: tests/Android.mk [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+LOCAL_SHARED_LIBRARIES := \
+        libutils \
+        liblog \
+        libc++ \
+        libva \
+        libva-android \
+        libui \
+        libgui \
+        libstagefright_foundation \
```
Comment (xuguangxin) [inline tests/Android.mk]: why we need this?

**Comment by xuguangxin** [comment]
@halleyzhao ,please +1 if you think it's ok.

**Comment by halleyzhao** [comment]
I'm  ok with the first two patch.

while the latest one looks strange to me: yami (even its client) calls directly to ufo driver, it is really ugly. 
since ufo driver tracks the pitch data of an AndroidNativeBuffer, then it is not necessary to force androidplayer to fill it with such tricky.
it seems bug/bad-design of ufo driver; you may contact the ufo driver developers for more information.

anyway, android-player is test only; you can select to merge it first.

**Comment by halleyzhao** [comment]
+1

**Comment by xuguangxin** [comment]
+1

**Comment by xuguangxin** [comment]
merged as 7ead058a66f0b188ede7bf2744af1797e2dfdbb5

**Pull request closed without merging**